File: bot.js
const TelegramBot = require('node-telegram-bot-api');

// Токен бота (вставь свой токен)
const token = '7836922267:AAEh99TfVahcs-Tfoqy8H8IYdfhprCIqWZ8';
const bot = new TelegramBot(token, { polling: true });

// URL вашего сайта (вставь свой URL)
const siteUrl = 'https://benzo-eight.vercel.app';

// Обработка сообщений
bot.on('message', (msg) => {
    const chatId = msg.chat.id;
    const text = msg.text;

    // Проверяем, является ли сообщение ссылкой
    if (text.match(/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/)) {
        // Отправляем пользователю ссылку на сайт с параметром
        const trackUrl = `${siteUrl}?track=${encodeURIComponent(text)}`;
        bot.sendMessage(chatId, `Слушайте трек: ${trackUrl}`);
    } else {
        bot.sendMessage(chatId, 'Пожалуйста, отправьте ссылку на трек.');
    }
});

File: bot.py
from telegram import Update, ReplyKeyboardMarkup
from telegram.ext import ApplicationBuilder, CommandHandler, MessageHandler, filters, CallbackContext
import requests

# Замени на свой API-токен
BOT_TOKEN = '7836922267:AAEh99TfVahcs-Tfoqy8H8IYdfhprCIqWZ8'

# URL сервера (пока локальный)
SERVER_URL = 'http://localhost:5000'

async def start(update: Update, context: CallbackContext):
    await update.message.reply_text(
        "Привет! Я бот для синхронизации музыки. Используй /play чтобы начать."
    )

async def play(update: Update, context: CallbackContext):
    # Отправляем состояние плеера
    response = requests.get(f'{SERVER_URL}/get_state')
    state = response.json()
    await update.message.reply_text(
        f"Сейчас играет: {state['current_track']}\n"
        f"Состояние: {'Играет' if state['is_playing'] else 'На паузе'}"
    )

async def handle_message(update: Update, context: CallbackContext):
    # Обработка ссылок или файлов
    if update.message.audio:
        file_id = update.message.audio.file_id
        await update.message.reply_text(f"Аудио получено! ID: {file_id}")
    elif update.message.text and 'soundcloud.com' in update.message.text:
        await update.message.reply_text("Ссылка на SoundCloud получена!")
    else:
        await update.message.reply_text("Я не понимаю эту команду.")

if __name__ == '__main__':
    app = ApplicationBuilder().token(BOT_TOKEN).build()

    # Регистрация обработчиков команд
    app.add_handler(CommandHandler("start", start))
    app.add_handler(CommandHandler("play", play))
    app.add_handler(MessageHandler(filters.ALL, handle_message))

    # Запуск бота
    app.run_polling()